homeassistant/components/radarr/config_flow.py

```diff
@@ -0,0 +1,187 @@
+"""Config flow for Radarr."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiohttp import ClientConnectorError
+from aiopyarr import exceptions
+from aiopyarr.models.host_configuration import PyArrHostConfiguration
+from aiopyarr.radarr_client import RadarrClient
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_HOST,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_URL,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import (
+    CONF_UPCOMING_DAYS,
+    DEFAULT_NAME,
+    DEFAULT_UPCOMING_DAYS,
+    DEFAULT_URL,
+    DOMAIN,
+    LOGGER,
+)
+
+
+class RadarrConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Radarr."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the flow."""
+        self.entry: ConfigEntry | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: ConfigEntry) -> OptionsFlow:
+        """Get the options flow for this handler."""
+        return RadarrOptionsFlowHandler(config_entry)
+
+    async def async_step_reauth(self, _: dict[str, str | bool]) -> FlowResult:
+        """Handle configuration by re-auth."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> FlowResult:
+        """Confirm reauth dialog."""
+        if user_input is not None:
+            return await self.async_step_user()
+
+        self._set_confirm_only()
+        return self.async_show_form(step_id="reauth_confirm")
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is None:
+            user_input = dict(self.entry.data) if self.entry else None
+
```

Comment:
```suggestion
```

Reply:
This and above suggestion fails this PR's current tests